src/utils/charconv.hpp

```diff
@@ -0,0 +1,88 @@
+/*
+	Copyright (C) 2024
+	Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY.
+
+	See the COPYING file for more details.
+*/
+
+#pragma once
+
+#include <boost/version.hpp>
+#include <string_view>
+#include <cctype>
+
+// The gcc implemenetation of from_chars is in some versions just a temporaty solution that calls
+// strtod that's why we prefer the boost version if available.
+#if BOOST_VERSION >= 108500
+	#define USE_BOOST_CHARCONV
+#elif defined(_GLIBCXX_RELEASE) && _GLIBCXX_RELEASE >= 11
+	#define USE_STD_CHARCONV
+#elif defined(_MSC_VER) && _MSC_VER >= 1924
+	#define USE_STD_CHARCONV
+#elif defined(_LIBCPP_VERSION) && _LIBCPP_VERSION >= 14000
+	#define USE_FALLBACK_CHARCONV
+#else
+	#error No charconv implementation found.
+#endif
+
+#ifdef USE_BOOST_CHARCONV
+#include <boost/charconv.hpp>
+#define CHARCONV_NAMESPACE boost::charconv
+#else
+#include <charconv>
+#define CHARCONV_NAMESPACE std
```

Comment:
I think it would be nicer if these were namespace aliases instead… unless the point is that you want to `#undef` them at the end of the file.

Namespace aliases means (if I recall correctly):

```c++
namespace ns = std;
```

Reply:
Sure, forgot they were a thing.